frontend/apps/filemanager/filemanagermenu.lua

```diff
@@ -889,10 +889,10 @@ function FileManagerMenu:_getTabIndexFromLocation(ges)
     if not ges then
         return last_tab_index
     -- if the start position is far right
-    elseif ges.pos.x > 2 * Screen:getWidth() / 3 then
+    elseif ges.pos.x > Screen:getWidth() * (2/3) then
```

Comment:
The bigger question's what they are on ARM than on x86 though. (Not an objection. :-) )

Reply:
Depends on the CPU (some... don't even *have* native euclidean div hardware ;)).

But generally the same principles apply (i.e., it's a few clock slower).

Which makes getting these out of the ARM technical manual a kind of a bitch, because it's *optional* on the A7 & A8 (IIRC, it's always supported on Kindle & Kobo, though).

But, if we look a ta slightly more modern processor where it's stock: https://hardwarebug.org/2014/05/15/cortex-a7-instruction-cycle-timings/

(So, yeah, much slower ;)).

Reply:
In compiled languages, it'd be the compiler's job to transforms as many divisions as possibly in MULs or shifts, but I'm not quite sure what happens in Lua's case ;o).

Reply:
LuaJIT tends to be pretty good about these things but you'd have to check. :-)

Reply:
I have made a small benchmark

```
x = 0
for i = 1,500000 do
    for j = 1,1000 do
        x = x + (j + i) * (1/3)
    end
end

print(x)
```
and
```
x = 0
for i = 1,500000 do
    for j = 1,1000 do
        x = x + (j + i) / 3
    end
end

pint(x)
```

Run both with `time luajit bench.lua`. 
On a Sage the first one takes `5.4s` the second one `12.4s`.
On my Laptop the first on needs `0.5s` the second one `2.1s`.

Reply:
> LuaJIT tends to be pretty good about these things but you'd have to check. :-)

In https://luapower.com/luajit-notes#luajit-assumptions there is
```
divisions are 4x slower than multiplications on x86, so when dividing by a constant, it helps turning x / c into x * (1 / c) since the constant expression is folded – LuaJIT does this already for power-of-2 constants where the semantics are equivalent.
```

Reply:
For context re: my link above, the Sage is running on a quad-A7 @ 1.8GHz, FWIW.

Our other devices tend to run on a A8 or an A9 at 1GHz.

Reply:
Cool, thanks.